temporalio/client.py

```diff
@@ -6114,6 +6195,142 @@ def _to_proto(self) -> temporalio.api.enums.v1.TaskReachability.ValueType:
             )
 
 
+class TaskQueueType(IntEnum):
+    """Type of task queue.
+
+    See :py:class:`temporalio.api.enums.v1.TaskQueueType`.
+    """
+
+    WORKFLOW = int(temporalio.api.enums.v1.TaskQueueType.TASK_QUEUE_TYPE_WORKFLOW)
+    ACTIVITY = int(temporalio.api.enums.v1.TaskQueueType.TASK_QUEUE_TYPE_ACTIVITY)
+    NEXUS = int(temporalio.api.enums.v1.TaskQueueType.TASK_QUEUE_TYPE_NEXUS)
+
+
+@dataclass
+class TaskQueueDescription:
+    """Description of a task queue."""
+
+    types: Mapping[TaskQueueType, TaskQueueTypeInfo]
```

Comment:
I'd prefer the following.
```suggestion
    unversioned_types: Mapping[TaskQueueType, TaskQueueTypeInfo]
```

Reply:
My reasoning is as follows.

The following diagram shows different "levels of task queue". ["Task Queue" is a overloaded term and it's been used to refer to any of the boxes in this diagram! In the server code at least we've refactored things and gave each box a different name.]

Now, from users perspective, Task Queue almost alway is the most top-level, black box. So DescribeTaskQueue, without any other specification, should describe that box.

Unversioned is just a slice of the whole Task Queue. A top-level field such as `types` is reasonable only if it maps to all slices (i.e. unversioned + build IDs). As server does not return all slices (and we rely on user specify the slices as of now), we cannot have a such top-level field. Maybe we can add it in the future.

As long as only unversioned stats are desired, it's totally fine, IMO, to put it in the top level proto but it should be qualified as such so user does not interpret it as something that holds the stats for all slices.

<img width="1452" alt="image" src="https://github.com/user-attachments/assets/99193e75-01f2-42f7-a710-395f39acedfa">

cc @cretz @Sushisource @dnr

Reply:
Agree

Reply:
I think people not using versioning should not have to know anything about versioning or have the term versioning in their code anywhere. Can we pretend these stats were made available before versioning was a thing (which is kinda the case since versioning is not available)?

Reply:
I think we should call things what they really are. We cannot pretend versioning is not there because it is. Even if it is pre-release, we know it's going to be public-preview and GA eventually.

If we call this only `types` it'll be dangerous or confusing for people who use versioning.

In your proposal as well you do have a note that talks about versioning and unversioned, so it's not like those terms will be completely absent from developers mind.

Lastly, I'm not sure if not having the term "unversioned" in the code anywhere for users not using versioning should be a goal TBH. I agree that they should not have to use term "version" or "build ID" in their code though.

Reply:
> I think we should call things what they really are

I don't expect we're going to change `Worker` to `UnversionedWorker`. We accept that the default is unversioned and we don't infect the default with the word unversioned anywhere. I do not think we should inconsistently start doing that.

> If we call this only types it'll be dangerous or confusing for people who use versioning.

I think we can document this away.